### ISSUE TITLE
Metadata support collecting table type

### ISSUE THREAD

**Issue by Pace2Car** [apache/shardingsphere#29816]
Now, table metadata does not collect table type.

Taking MySQL as the data source as an example, when executing `show full tables` via Proxy, the result is:
```sql
mysql> show full tables;
+-----------------------+------------+
| Tables_in_sharding_db | Table_type |
+-----------------------+------------+
| t_order               | BASE TABLE |
| t_view                | BASE TABLE |
+-----------------------+------------+

```
When executing directly from the MySQL client, the result is:
```sql
mysql> show full tables;
+---------------------+------------+
| Tables_in_demo_ds   | Table_type |
+---------------------+------------+
| t_order             | BASE TABLE |
| t_view              | VIEW       |
+---------------------+------------+
```

I plan to enhance the metadata, collect table type to support get the correct table type when executing `show full tables`

Task list:
- [x] TableMetaData add column tableType
- [x] DialectMetaDataLoader‘s implementation classes implements collection table type
  - [x] H2MetaDataLoader
  - [x] PostgreSQLMetaDataLoader
  - [x] MySQLMetaDataLoader
  - [x] OracleMetaDataLoader
  - [x] OpenGaussMetaDataLoader
  - [x] SQLServerMetaDataLoader
- [x] UT/IT cases

**Comment by zhaojinchao95** [comment]
@Pace2Car Hi, the `5.5.0` version will be released next week. Can it be completed during this period?

**Comment by Pace2Car** [comment]
> @Pace2Car Hi, the `5.5.0` version will be released next week. Can it be completed during this period?

Remaining tasks can be postponed

**Comment by terrymanu** [comment]
Hi @Pace2Car, thanks for tracking this. The core part has already landed in master, but I'd keep this issue open until the remaining dialects and tests are done.

Already merged:
- `TableType` was added to table metadata, and MySQL collection was implemented in #29953, where `MySQLMetaDataLoader` marks views via `information_schema.VIEWS`.
- The PostgreSQL, Oracle and OpenGauss loaders now also set `TableType.VIEW` for views on master.

Still missing on master:
- `H2MetaDataLoader` and `SQLServerMetaDataLoader` still build `TableMetaData` with the constructor that defaults to `TableType.TABLE`, so views are still loaded as tables for these two databases.
- The UT/IT item is not covered yet: `MySQLShowTablesExecutorTest#assertShowTablesExecutorWithFull` only asserts the column count, and no test asserts the `Table_type` value returned by `SHOW FULL TABLES`.
- Worth aligning at the same time: the proxy renders the enum name, so `SHOW FULL TABLES` returns `TABLE` for base tables while native MySQL returns `BASE TABLE`.

So I suggest keeping this issue open with `status: volunteer wanted` for the H2 and SQLServer loaders plus the missing test assertions. Once a PR covering them is merged, we can close this issue as completed. I can tick the finished checklist items in the description.
